### PR TITLE
[#8787] New instructors: set the time zone of the sample course to match user

### LINKED ISSUE #8787: New instructors: set the time zone of the sample course to match user
Steps:
1. Admin creates a new instructor
2. New instructor joins TEAMMATES
3. Upon joining, new instructor is given a sample course 

Current: sample course is in UTC timezone
Suggested: set the timezone to user's time zone

Reason: Sample curse is used to create a first impression, it's nice if the time zone is set correctly.

### PR DESCRIPTION
**Pull request by samuelfangjw**
Fixes #8787 

**Outline of Solution**

1. In `account.service.ts`, `createAccount()` has a new parameter `timezone`.
2. When a user creates an account, timezone is passed to as an argument to `createAccount()`. The timezone is guessed from the user's web browser settings. Timezone is sent as part of the post request.
3. In `CreateAccountAction`, when creating demo course, timezone in the template is replaced by the user's timezone. If timezone if invalid, default timezone is used instead.
4. The time is adjusted to take into account timezone. For example, all sessions are adjusted to open and close at  midnight in the new timezone.

**Points to note**
I don't believe `CreateAccountAction#replaceAdjustedTimeAndTimezone()` can throw any error under normal execution conditions. Not sure if there is any runtime error to catch here.

### CODE REVIEW

**Comment by wkurniawan07** [comment]
Quoting myself from https://github.com/TEAMMATES/teammates/pull/11243#issuecomment-898993617:
> In case you're not aware, the sessions of the sample course are set to have end time at 11.59pm in a preset timezone. Moving the timezone without also moving those end times will make it really weird, e.g. "why is this sample session ending at 4am?".

**Comment by samuelfangjw** [comment]
> Quoting myself from [#11243 (comment)](https://github.com/TEAMMATES/teammates/pull/11243#issuecomment-898993617):
> 
> > In case you're not aware, the sessions of the sample course are set to have end time at 11.59pm in a preset timezone. Moving the timezone without also moving those end times will make it really weird, e.g. "why is this sample session ending at 4am?".

oh right, will update.

**Review by NicolasCwy — CHANGES_REQUESTED** [review]
Mostly nits, but also should `InstructorSampleData.json` have a working default state, commented directly on that section!

File: src/main/resources/InstructorSampleData.json [hunk 1]
```diff
@@ -5,7 +5,7 @@
       "id": "demo.course",
       "name": "Sample Course 101",
       "institute": "demo.institute",
-      "timeZone": "UTC"
+      "timeZone": "demo.timezone"
```
[the remark this answers is not in the document]
Comment (samuelfangjw) [reply src/main/resources/InstructorSampleData.json:8]: Thanks for spotting this

**Comment by samuelfangjw** [comment]
@NicolasCwy Made the suggested changes! Also added a check in `CreateAccountAction#importDemoData()` - if timezone is default timezone (UTC), no need to replace timezone specific data. Could you take another look to see if it's ok?

**Review by NicolasCwy — APPROVED** [review]
LGTM!

**Review by wkurniawan07 — APPROVED** [review]

**Pull request merged**